src/index.ts

```diff
@@ -245,7 +247,7 @@ export class ContextListener<T> {
 
   /* called by provider */
   onDisconnect = () => {
-    this.status = ListenerConnectionStatus.CONNECTING;
+    this._status = ListenerConnectionStatus.CONNECTING;
```

Comment:
Unintentional change, won't notify

```suggestion
    this.status = ListenerConnectionStatus.CONNECTING;
```